### DISCUSSION TITLE
How does Mpool manage two layers of devices?

### DISCUSSION THREAD

**Discussion by sherlockwu** [hse-project/mpool#2] [Q&A]
Hi Mpool team,

Thanks for putting your project on GitHub!
Could you help to clarify something to help understand what Mpool target for?

1. Looks like Mpool is indeed the module that does heterogeneous devices management for the HSE key-value store project, is this correct?

2. How does Mpool manage two layers of devices? Is it caching on the faster layer? Could you provide some description on this? thanks

Kan

**Comment by smoyerx** [comment]
Please see the **Concepts** section of the Wiki for a high-level overview.

In short, mpool itself simply provides the ability to allocate an object in one of the configured media classes (capacity or staging).  In the context of HSE (which is to say when HSE is the mpool client), it is HSE that decides in which media class to allocate an object based on what it stores.

**Comment by sherlockwu** [comment]
I see, that makes sense. I will head to look into HSE then. 
Thanks much
